business/openid_auth.go

```diff
@@ -278,17 +297,9 @@ func GetOpenIdMetadata() (*OpenIdMetadata, error) {
 		// Remove trailing slash from issuer URI, if needed
 		trimmedIssuerUri := strings.TrimRight(cfg.IssuerUri, "/")
 
-		// Create HTTP client
-		httpTransport := &http.Transport{}
-		if cfg.InsecureSkipVerifyTLS {
-			httpTransport.TLSClientConfig = &tls.Config{
-				InsecureSkipVerify: true,
-			}
-		}
-
-		httpClient := http.Client{
-			Timeout:   time.Second * 10,
-			Transport: httpTransport,
+		httpClient, err := createHttpClient(trimmedIssuerUri)
+		if err != nil {
+			return nil, err
```

Comment:
Can you, please, return a descriptive error alike in the other error checks?
Like "cannot create HTTP client to fetch metadata"? (plus the inner error).